CodeGeneration/Models+Copiable.swifttemplate

```diff
@@ -0,0 +1,139 @@
+<%# 
+//
+// Generates .copy methods for all structs and classes conforming to the `GeneratedCopiable` protocol
+//
+-%>
+<% 
+// -------------------------------------------------------------------------------- 
+// Utils
+// -------------------------------------------------------------------------------- 
+
+/// Guess the `moduleName` of a `TypeName` given by Sourcery.
+///
+/// Based on some experiments, built-in types like `UINotificationFeedbackGenerator.FeedbackType` are 
+/// not given a `.module` property value by Sourcery. Which makes sense, I guess?
+///
+/// - SeeAlso: https://cdn.rawgit.com/krzysztofzablocki/Sourcery/master/docs/Classes/Type.html
+///
+func guessModuleOf(typeName: TypeName) -> String? {
+    if (typeName.name.hasPrefix("UI")) {
+        return "UIKit"
+    } else {
+        return nil
+    }
+}
+
+-%>
+<%
+// -------------------------------------------------------------------------------- 
+// Gather Information
+// -------------------------------------------------------------------------------- 
+
+// The module where the CopiableProp and NullableCopiableProp typealiases belong.
+let copiablePropModule = "Networking"
+
+// The matching types that we're going to generate code for.
+let matchingTypes = types.based["GeneratedCopiable"].filter { 
+    $0.kind == "struct" || $0.kind == "class"
+}
+
+// The names of modules that we should generate "import MODULE_NAME" lines for.
+let modulesToGenerateImports: [String] = { 
+    let modulesFromProperties: [String] = matchingTypes.flatMap { type in 
+        type.variables.compactMap { variable in 
+            if let variableModule = variable.type?.module {
+                return variableModule
+            }
+
+            return guessModuleOf(typeName: variable.typeName)
+        }
+    }
+
+    let modulesRequiredByTemplate: [String] = {
+        let shouldImportCopiablePropModule = matchingTypes.contains(where: { $0.module != copiablePropModule })
+        return shouldImportCopiablePropModule ? [copiablePropModule] : []
+    }()
+
+    return Array(Set(modulesFromProperties + modulesRequiredByTemplate)).sorted().filter {
+        // Ignore modules that belong to the current types we're generating for.
+        $0 != matchingTypes.first?.module
+    }
+}()
+
+/// Defines the properties that the template will need to generate the copy() method for a single type.
+///
+/// We create our own data structure to clarify what we need in the template code below. 
+/// This also makes the template more simpler to read because the complexity are all encapsulated
+/// by this struct.
+///
+struct CopiableSpec {
+    struct Property {
+        let name: String
+        /// String, Int, etc 
+        let typeName: String
+        /// NullableCopiableProp or CopiableProp
+        let copiablePropTypeName: String
+        /// If this is not the last, this will be a literal comma (",")
+        let commaOrNothing: String
+    }
+
+    let name: String
```

Comment:
it took me some time to understand the `CopiableSpec` struct - this name here is the name of the class/struct that conforms to `GeneratedCopiable` right?

Reply:
the comments are helpful, thanks for adding them!